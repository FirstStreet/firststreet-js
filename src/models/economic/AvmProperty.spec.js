const _ = require('lodash');
const Sut = require('./AvmProperty.js');

const mock = require('../../__mocks__/avmProperty.json');

describe('AVM property', () => {
  it('should construct object', () => {
    const result = new Sut(mock);
    expect(result.raw).toMatchSnapshot();
    expect(result.avm).toBeDefined();
    expect(result.avm).toMatchSnapshot();
    expect(result.avm.low).not.toBeDefined();
    expect(result.avm.mid).toEqual(144500);
    expect(result.fsid).toEqual(4801470191);
  });
});

describe('AVM property', () => {
  it('should construct when there\'s no AVM', () => {
    let mockCopy = _.clone(mock);
    mockCopy = _.omit(mockCopy, ['avm']);
    const result = new Sut(mockCopy);
    expect(result.raw).toMatchSnapshot();
    expect(result.avm).not.toBeDefined();
    expect(result.fsid).toEqual(4801470191);
  });
});
