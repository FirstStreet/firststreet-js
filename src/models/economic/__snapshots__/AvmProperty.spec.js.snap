// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AVM property should construct object 1`] = `
Object {
  "avm": Object {
    "mid": 144500,
  },
  "fsid": 4801470191,
  "providerID": 2,
}
`;

exports[`AVM property should construct object 2`] = `
Avm {
  "_avmRecord": Object {
    "mid": 144500,
  },
}
`;

exports[`AVM property should construct when there's no AVM 1`] = `
Object {
  "fsid": 4801470191,
  "providerID": 2,
}
`;
